state/state.go

```diff
@@ -278,6 +259,10 @@ type Server struct {
 
 	SelfTest *SelfTestResult
 
+	WebSocket      *websocket.Conn
+	SnapshotStream chan []byte
+	Pause          pganalyze_collector.ServerMessage_Pause
```

Comment:
These probably need a synchronization primitive, though I wasn't sure which one is the best fit, and this PR adds data-race detection to the `make run` command in the Makefile which didn't find an issue here.

Reply:
Oof, the existing races are a mess. I wonder if we shouldn't rethink our flow rather than try to fix those directly. No strong feelings on the new mechanism.